napari/settings/_tests/test_settings.py

```diff
@@ -300,6 +301,41 @@ def test_get_settings(tmp_path):
     assert str(s.config_path) == str(p)
 
 
+def test_get_prev_ver_settings(monkeypatch, tmp_path):
+    """This test tests fallback to previous version settings"""
+    import napari.utils._appdirs
+
+    # set the current version to 0.4.19 for testing
+    monkeypatch.setattr(napari.utils._appdirs, 'version_string', '0.4.19')
+    # prep a settings file for the previous version
+    data = "appearance:\n   theme: light"
+    prev_path = tmp_path / '0.4.18' / 'settings.yaml'
+    prev_path.parent.mkdir(parents=True, exist_ok=True)
+    prev_path.write_text(data)
```

Comment:
Optional: you use this code block three times and it takes up the bulk of the test slightly obscuring the interesting stuff (i.e. the contents and versions), so consider refactoring it a helper/setup function.

```python
def write_settings_file(*, root: Path, version: str, contents: str) -> None:
    path = root / '0.4.18' / 'settings.yaml'
    path.parent.mkdir(parents=True, exist_ok=True)
    path.write_text(data)
```